extensions/amp-selector/1.0/selector.js

```diff
@@ -239,23 +241,31 @@ function SelectorWithRef(
   );
 
   return (
-    <Comp
-      {...rest}
-      role={role}
-      aria-disabled={disabled}
-      aria-multiselectable={multiple}
-      disabled={disabled}
-      keyboardSelectMode={keyboardSelectMode}
-      multiple={multiple}
-      onKeyDown={onKeyDown}
-      tabIndex={
-        tabIndex ?? keyboardSelectMode === KEYBOARD_SELECT_MODE.SELECT ? 0 : -1
-      }
-    >
-      <SelectorContext.Provider value={context}>
-        {children}
-      </SelectorContext.Provider>
-    </Comp>
+    <>
+      <Comp
+        {...rest}
+        role={role}
+        aria-disabled={disabled}
+        aria-multiselectable={multiple}
+        disabled={disabled}
+        form={form}
+        keyboardSelectMode={keyboardSelectMode}
+        multiple={multiple}
+        name={name}
+        onKeyDown={onKeyDown}
+        tabIndex={
+          tabIndex ?? keyboardSelectMode === KEYBOARD_SELECT_MODE.SELECT
+            ? 0
+            : -1
+        }
+        value={selected}
+      >
+        <SelectorContext.Provider value={context}>
+          {children}
+        </SelectorContext.Provider>
+      </Comp>
+      <input hidden defaultValue={selected} name={name} form={form} />
```

Comment:
I feel like it'd be better to put the `input` hidden inside the `<Comp>`. It's also more symmetric with the AMP implementation. I'd place it as a first. I definitely see you point on the `:first-/:last-child`, but I think those are more relevant in the custom element case, and less relevant in React-style app.

Reply:
Okay, moved inside.